app/forms/node_form.rb

```diff
@@ -59,10 +68,14 @@ def create_objects
   def update_objects
     ActiveRecord::Base.transaction do
       node = Node.find(id)
-      node.update_attributes!(name: name, cost_code: cost_code, description: description, parent_id: parent_id)
-      node.permissions.destroy_all
-      node.set_permissions
-      node.permissions.create!(convert_permissions(node.owner))
+      if node.update_attributes(name: name, cost_code: cost_code, description: description, parent_id: parent_id)
+        node.permissions.destroy_all
+        node.set_permissions
+        node.permissions.create!(convert_permissions(node.owner))
+      else
+        @node = node
```

Comment:
Again, this isn't necessary. Also, `@node` won't be set if anything other than `update_attributes` fails.

I'd put it back to the way it was but like the other example, the `node` could instead be an instance variable e.g. `@node = Node.find(id)`